docs/migration.md

```diff
@@ -0,0 +1,193 @@
+  flags relating to the old engine have been removed.  For information on the
+  new enginer, see [Storage](#storage).
+
+- `-storage.remote.*` Prometheus 2.0 has removed the deprecated remote storage
+  flags.  To write to InfluxDB, Graphite, or OpenTSDB use the storage adapter.
+
+## Config File
+
+## Alertmanager Service Discovery
+
+Alertmanager service discovery was introduced in Prometheus 1.4, allowing Prometheus
+to dynamically discover Alertmanager replicas using the same mechanism as monitoring
+targets.  In Prometheus 2.0, the command line flags for static Alertmanager config
+have been removed, so the following command line flag:
+
+```
+./prometheus -alertmanager.url=http://alertmanager/
+```
+
+Might looks like the following snippet in the `prometheus.yml` config file:
+
+```yml
+alerting:
+  alertmanagers:
+  - static_configs:
+    - targets:
+      - alertmanager
+```
+
+But note, you can use all the usual Prometheus service discovery integrations
+and relabelling.  In this snippet I'm instructing Prometheus
+to search for Kubernetes pods, in the `default` namespace, with the label
+`name: alertmanager` and with a non-empty port.
+
+```yml
+alerting:
+  alertmanagers:
+  - kubernetes_sd_configs:
+      - role: pod
+    tls_config:
+      ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+    bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
+    relabel_configs:
+    - source_labels: [__meta_kubernetes_pod_label_name]
+      regex: alertmanager
+      action: keep
+    - source_labels: [__meta_kubernetes_namespace]
+      regex: default
+      action: keep
+    - source_labels: [__meta_kubernetes_pod_container_port_number]
+      regex:
+      action: drop
+```
+
+## Recording rules an alerts
+
+The format for configuring alerting and recoding rules has been changed to YAML.
+An example of an recording rule in the old format:
+
+```
+job:request_duration_seconds:99percentile =
+  histogram_quantile(0.99, sum(rate(request_duration_seconds_bucket[1m])) by (le, job))
+
+ALERT FrontendRequestLatency
+  IF job:request_duration_seconds:99percentile{job="frontend"} > 0.1
+  FOR 5m
+  ANNOTATIONS {
+    summary = "High frontend request latency",
+  }
+```
+
+Might looks like this in the new format:
+
+```yml
+groups:
+- name: example.rules
+  rules:
+  - record: job:request_duration_seconds:99percentile
+    expr: histogram_quantile(0.99, sum(rate(request_duration_seconds_bucket[1m]))
+      BY (le, job))
+  - alert: FrontendRequestLatency
+    expr: job:request_duration_seconds:99percentile{job="frontend"} > 0.1
+    for: 5m
+    annotations:
+      summary: High frontend request latency
+```
+
+The `promtool` has a command to automate the conversion.  Given a `.rules` file,
+it will output a `.rules.yml` file in the new format. For example:
+
+```
+$ promtool update rules example.rules
+```
+
+## Storage
+
+The data format in Prometheus 2.0 has completely changed and is not backwards
```

Comment:
Nit, but "backward compatible" was used previously instead of "backwards compatible".

Reply:
They all seem to be "backwards"

Reply:
backwards it is.